common/src/main/java/io/seata/common/util/IdWorker.java

```diff
@@ -14,161 +14,173 @@
+        long timestampWithSequence = timestamp << sequenceBits;
+        this.timestampAndSequence = new AtomicLong(timestampWithSequence);
+    }
 
     /**
-     * Constructor
-     *
-     * @param workerId
-     *            Job ID (0 ~ 1023)
+     * init workerId
+     * @param workerId if null, then auto generate one
      */
-    public IdWorker(long workerId) {
+    private void initWorkerId(Long workerId) {
+        if (workerId == null) {
+            workerId = generateWorkerId();
+        }
         if (workerId > maxWorkerId || workerId < 0) {
-            throw new IllegalArgumentException(
-                String.format("worker Id can't be greater than %d or less than 0", maxWorkerId));
+            String message = String.format("worker Id can't be greater than %d or less than 0", maxWorkerId);
+            throw new IllegalArgumentException(message);
         }
-        this.workerId = workerId;
+        this.workerId = workerId << (timestampBits + sequenceBits);
     }
 
     /**
-     * Get the next ID (the method is thread-safe)
-     *
-     * @return SnowflakeId
+     * get next UUID(base on snowflake algorithm), which look like:
+     * highest 1 bit: always 0
+     * next   10 bit: workerId
+     * next   41 bit: timestamp
+     * lowest 12 bit: sequence
+     * @return UUID
      */
-    public synchronized long nextId() {
-        long timestamp = timeGen();
-
-        if (timestamp < lastTimestamp) {
-            throw new RuntimeException(String.format(
-                "clock moved backwards.  Refusing to generate id for %d milliseconds", lastTimestamp - timestamp));
-        }
-
-        if (lastTimestamp == timestamp) {
-            sequence = (sequence + 1) & sequenceMask;
-            if (sequence == 0) {
-                timestamp = tilNextMillis(lastTimestamp);
-            }
-        } else {
-            sequence = 0L;
-        }
-        lastTimestamp = timestamp;
-
-        return ((timestamp - twepoch) << timestampLeftShift) | (workerId << workerIdShift) | sequence;
+    public long nextId() {
+        waitIfNecessary();
+        long next = timestampAndSequence.incrementAndGet();
+        long timestampWithSequence = next & timestampAndSequenceMask;
+        return workerId | timestampWithSequence;
     }
 
     /**
-     * Block until the next millisecond until a new timestamp is obtained
-     *
-     * @param lastTimestamp
-     *            Time of last ID generation
-     * @return Current timestamp
+     * block current thread if the QPS of acquiring UUID is too high
+     * that current sequence space is exhausted
      */
-    protected long tilNextMillis(long lastTimestamp) {
-        long timestamp = timeGen();
-        while (timestamp <= lastTimestamp) {
-            timestamp = timeGen();
+    private void waitIfNecessary() {
+        long currentWithSequence = timestampAndSequence.get();
+        long current = currentWithSequence >>> sequenceBits;
```

Comment:
if currentWithSequence will never be negative number, I think user
```
>>
```
is enought.

Reply:
currentWithSequence could be a negative number(though it would be a long long time to reach this~). What ever, >>> and >> have the same speed for cpu.